### PR TITLE
Add namespace test in destination acceptance test

### LINKED ISSUE #9351: Create a well defined charset for namespaces and test them in DAT
## Tell us about the problem you're trying to solve
issues like https://github.com/airbytehq/airbyte/issues/7435 happen because we don't have clear rules for what characters should be allowed in a namespace. In addition, destinations don't have clear guidance around how to handle namespaces which contain special characters e.g: a custom namespace with a `-` or a space could cause a destination to fail. 

## Describe the solution you’d like
1. define exactly what characters / rules are allowed in a namespace (both custom and source-defined namespaces).
1. Destinations should be resilient towards all allowed characters
2. DAT should test that all destinations can handle all namespace chars. For this one it is super important that we put on the chaos monkey hat in choosing namespace values in the test to make sure the destination can handle everything we throw at it.

### PR DESCRIPTION
**Pull request by tuliren**
## What
- Close https://github.com/airbytehq/airbyte/issues/9351.
- The new versions are published in https://github.com/airbytehq/airbyte/pull/11274.

## How
- Add a new test in DAT.
- Run a simple data sync to a variety of namespaces on the destination.
- The test is disabled by default. It is currently only activated for Snowflake and BigQuery.

## Recommended reading order
1. `DestinationAcceptanceTest.java`
2. `namespace_test_cases.json`
3. `SnowflakeInternalStagingDestinationAcceptanceTest.java`

## 🚨 User Impact 🚨
- None expected.

### CODE REVIEW

File: airbyte-integrations/bases/standard-destination-test/src/main/resources/namespace_test_cases.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
+[
+  {
+    "id": "s1-1",
```
Comment (edgao) [inline airbyte-integrations/bases/standard-destination-test/src/main/resources/namespace_test_cases.json:3]: nitpick: can these IDs be a little more descriptive (e.g. `all_caps`) to make them easier to cross-reference? right now if a test fails, it'll just log that case `s1-1` failed, and then  the developer would need to find this file to understand why.

**Review by edgao — APPROVED** [review]
:shipit:

File: airbyte-integrations/bases/standard-destination-test/src/main/resources/namespace_test_cases.json (hunk 1 above)
Comment (tuliren) [reply airbyte-integrations/bases/standard-destination-test/src/main/resources/namespace_test_cases.json]: Done. I also updated the exception message to ask people to look into the `namespace_test_case.json` for details.

**Pull request merged**